runtime/src/bank.rs

```diff
@@ -2972,11 +2971,11 @@ mod tests {
 
         let bank1 = new_from_parent(&bank);
         assert_eq!(bank1.is_delta.load(Ordering::Relaxed), false);
-        assert_eq!(bank1.hash_internal_state(), bank.hash());
+        assert_ne!(bank1.hash_internal_state(), bank.hash());
         // ticks don't make a bank into a delta
         bank1.register_tick(&Hash::default());
         assert_eq!(bank1.is_delta.load(Ordering::Relaxed), false);
-        assert_eq!(bank1.hash_internal_state(), bank.hash());
+        assert_ne!(bank1.hash_internal_state(), bank.hash());
```

Comment:
nits:

This assertion would succeed even if register_tick din't change internal state.
Maybe use last value like this?
```rust
         let hash1 = bank1.hash_internal_state();
         assert_ne!(hash1, bank.hash()); 
         ....
         bank1.register_tick(&Hash::default());
         ....
         assert_ne!(bank1.hash_internal_state(), hash1); 
```


Generally negation assertions tend to be false negatives quickly so we have to be careful when using it. :D

Reply:
You're right, that's a excellent point. I'll push an update.